File: java/meetup/src/main/java/Meetup.java
import java.time.DayOfWeek;
import java.time.LocalDate;

public class Meetup {

    private final int month, year;

    public Meetup(int month, int year) {
        this.month = month;
        this.year  = year;
    }

    public LocalDate day(DayOfWeek dayOfWeek, MeetupSchedule meetupSchedule) {
//        LocalDate date = LocalDate.of(year, month, 1);

        switch(meetupSchedule){
            case FIRST:{
                LocalDate date = LocalDate.of(year, month, 1);

                return date.plusDays((dayOfWeek.getValue() + 7 - date.getDayOfWeek().getValue()) % 7);
            }
            case SECOND: {
                LocalDate date = LocalDate.of(year, month, 8);

                return date.plusDays((dayOfWeek.getValue() + 7 - date.getDayOfWeek().getValue()) % 7);
            }
            case THIRD: {
                LocalDate date = LocalDate.of(year, month, 15);

                return date.plusDays((dayOfWeek.getValue() + 7 - date.getDayOfWeek().getValue()) % 7);
            }
            case FOURTH: {
                LocalDate date = LocalDate.of(year, month, 22);

                return date.plusDays((dayOfWeek.getValue() + 7 - date.getDayOfWeek().getValue()) % 7);
            }
            case LAST: {
                LocalDate date = LocalDate.of(year, month, 1).plusMonths(1).minusDays(1);

                return date.minusDays((date.getDayOfWeek().getValue() - dayOfWeek.getValue() + 7 ) % 7);
            }
            case TEENTH: {
                LocalDate date = LocalDate.of(year, month, 13);

                return date.plusDays((dayOfWeek.getValue() + 7 - date.getDayOfWeek().getValue()) % 7);
            }
        }

        return null;
    }
}

File: csharp/meetup/Meetup.cs
using System;

public enum Schedule
{
    Teenth,
    First,
    Second,
    Third,
    Fourth,
    Last
}

public class Meetup
{
    private readonly int year;
    private readonly int month;

    public Meetup(int month, int year)
    {
        this.year = year;
        this.month = month;
    }

    public DateTime Day(DayOfWeek dayOfWeek, Schedule schedule)
    {
        DateTime date = new DateTime(year, month,
            schedule switch
            {
                Schedule.First => 1,
                Schedule.Second => 8,
                Schedule.Third => 15,
                Schedule.Fourth => 22,
                Schedule.Last => DateTime.DaysInMonth(year, month) - 6,
                Schedule.Teenth => 13,
                _ => throw new ArgumentOutOfRangeException()
            }); 

        int daysFromRequiredDayOfWeek = (7 + dayOfWeek - date.DayOfWeek) % 7; 
        date = date.AddDays(daysFromRequiredDayOfWeek);

        return date;
    }
}